treeherder/perf/investigation_aids.py

```diff
@@ -0,0 +1,184 @@
+        actions_context = self._load(decision_task_id, task_id)
+        action_to_trigger = self._get_action(actions_context['actions'], action)
+
+        return self._submit(
+            action=action_to_trigger,
+            decision_task_id=decision_task_id,
+            task_id=task_id,
+            input=input,
+            static_action_variables=actions_context['staticActionVariables']
+        )
+
+    def _load(self, decision_task_id: str, task_id: Job) -> dict:
+        if not decision_task_id:
+            raise ValueError("No decision task, can't find taskcluster actions")
+
+        # fetch
+        logger.debug('Fetching actions.json...')
+        artifact_response = self.queue.getLatestArtifact(task_id=decision_task_id,
+                                                         name='public/actions.json')
+        task_definition = self.queue.task(task_id)
+
+        if not artifact_response.ok:
+            raise RuntimeError('Unable to load actions.json')
+
+        actions_json = artifact_response.json()
+        if actions_json['version'] != 1:
+            raise RuntimeError('Wrong version of actions.json, unable to continue')
+
+        return {
+            'staticActionVariables': actions_json['variables'],
+            'actions': self._filter_relevant_actions(actions_json, task_definition),
+        }
+
+    def _submit(self,
+                action=None,
+                decision_task_id=None,
+                task_id=None,
+                input=None,
+                static_action_variables=None) -> str:
+        context = {
+                "taskGroupId": decision_task_id,
+                "taskId": task_id or None,
+                "input": input,
+         }
+        context.update(static_action_variables)
+
+        if action["kind"] == "task":
+            raise NotImplementedError("Unable to submit actions with 'task' kind.")
+
+        if action["kind"] == "hook":
+            hook_payload = jsone.render(action["hookPayload"], context)
+            hook_id, hook_group_id = action["hookId"], action["hookGroupId"]
+
+            decision_task = self.queue.task(decision_task_id)
+            expansion = self.auth.expandScopes({"scopes": decision_task["scopes"]})
+            expression = f"in-tree:hook-action:{hook_group_id}/{hook_id}"
+
+            if not satisfiesExpression(expansion["scopes"], expression):
+                raise RuntimeError(f"Action is misconfigured: decision task's scopes do not satisfy {expression}")
+
+            result = self.hooks.triggerHook(hook_group_id, hook_id, hook_payload)
+            return result["status"]["taskId"]
+
+    @classmethod
+    def _filter_relevant_actions(cls, actions_json: dict, original_task) -> List:
+        actions = actions_json['actions']
+
+        def collect_relevant_actions(accumulating_actions, action):
+            acc_action_names = [action['name'] for action in accumulating_actions]
+
+            action_of_known_kind = cls._has_known_kind(action)
```

Comment:
I missed this before. You should *not* be checking the kinds here. 

This should find the first action of each name that has the right context. The kind of action should be irrelevant to that. Of course, if it is an unsupported kind, you won't be able to *do* anything with it. That is, if there are two entries that would match, you should pick the first one, even if the kind is unsupported.

`_submit` can then error if it finds an unknown kind.

---

In practice, this doesn't matter, since we only generate tasks with a kind of hook.